clang/lib/DPCT/ThrustAPIMigration.cpp

```diff
@@ -29,7 +29,13 @@ void ThrustAPIRule::registerMatcher(ast_matchers::MatchFinder &MF) {
                          hasDeclContext(namespaceDecl(hasName("thrust")))))))))
           .bind("thrustFuncCall"),
       this);
-
+  // THRUST_STATIC_ASSERT macro register
+  MF.addMatcher(staticAssertDecl(isExpandedFromMacro("THRUST_STATIC_ASSERT"))
+                    .bind("THRUST_STATIC_ASSERT"),
+                this);
+  MF.addMatcher(typedefDecl(isExpandedFromMacro("THRUST_STATIC_ASSERT"))
```

Comment:
Is the addMatcher call with typedefDecl necessary?  The testcase passes without it.